syntax = "proto3";

option csharp_namespace = "GrpcSandbox.Core.Protos";

import "google/protobuf/timestamp.proto";

service CustomerService {
	rpc GetCustomerInfo (CustomerLookupRequest) returns (Customer);
	rpc GetNewCustomers (NewCustomerRequest) returns (stream Customer);
}

message NewCustomerRequest {}

message CustomerLookupRequest {
	int32 UserId = 1;
}

message Customer {
	string FirstName = 1;
	string LastName = 2;
	string EmailAddress = 3;
	bool IsAlive = 4;
	int32 Age = 5;
	int32 Id = 6;
	Address BillingAddress = 7;
	Address ShippingAddress = 8;
	repeated string Receipts = 9;
	Membership Membership = 10;
	google.protobuf.Timestamp CreatedAt = 11;
	google.protobuf.Timestamp ModifiedAt = 12;
}

message Address {
	string Line1 = 1;
	string Line2 = 2;
	string Line3 = 3;
	string County = 4;
	string Province = 5;
	string PostCode = 6;
	string Country = 7;
}

enum Membership {
	Guest = 0;
	Basic = 1;
	Premium = 2;
}
